.github/workflows/main.yml

```diff
@@ -25,11 +25,14 @@ jobs:
           python-version: "${{ matrix.python-version }}"
           allow-prereleases: true
 
+      - uses: hynek/setup-cached-uv@v2
+      - uses: extractions/setup-just@v3
+
       - name: "Install dependencies"
         run: |
           python -VV
           python -m site
-          python -m pip install --upgrade pip wheel pdm
+          python -m pip install --upgrade pip wheel uv
           python -m pip install --upgrade tox tox-gh-actions
```

Comment:
It is the uv that must install dependencies, the current approach is not what uv official docs recommend concerning [GHA integration](https://docs.astral.sh/uv/guides/integration/github/).